src/cli/cli.ts

```diff
@@ -0,0 +1,1052 @@
+import Configstore from 'configstore';
+import fs from 'fs';
+import minimist from 'minimist';
+import { createRequire } from 'module';
+import ora, { Options, Ora } from 'ora';
+import os from 'os';
+import path from 'path';
+import { fileURLToPath, pathToFileURL } from 'url';
+import Command, { CommandArgs, CommandError, CommandTypes } from '../Command.js';
+import GlobalOptions from '../GlobalOptions.js';
+import config from '../config.js';
+import { M365RcJson } from '../m365/base/M365RcJson.js';
+import request from '../request.js';
+import { settingsNames } from '../settingsNames.js';
+import { telemetry } from '../telemetry.js';
+import { app } from '../utils/app.js';
+import { formatting } from '../utils/formatting.js';
+import { md } from '../utils/md.js';
+import { validation } from '../utils/validation.js';
+import { CommandInfo } from './CommandInfo.js';
+import { CommandOptionInfo } from './CommandOptionInfo.js';
+import { Logger } from './Logger.js';
+import { SelectionConfig, ConfirmationConfig, prompt } from '../utils/prompt.js';
+import { timings } from './timings.js';
+const require = createRequire(import.meta.url);
+
+const __dirname = fileURLToPath(new URL('.', import.meta.url));
+
+export interface CommandOutput {
+  stdout: string;
+  stderr: string;
+}
+
+let _config: Configstore | undefined;
+// we assign it through exported function to support mocking
+// eslint-disable-next-line prefer-const
+let spinner: Ora = ora();
+const commands: CommandInfo[] = [];
+/**
+ * Command to execute
+ */
+let commandToExecute: CommandInfo | undefined;
+/**
+ * Name of the command specified through args
+ */
+let currentCommandName: string | undefined;
+let optionsFromArgs: { options: minimist.ParsedArgs } | undefined;
+const defaultHelpMode = 'options';
+const helpModes: string[] = ['options', 'examples', 'remarks', 'response', 'full'];
+
+function getConfig(): Configstore {
```

Comment:
Shall we rename this to `getConfigStore`?

It makes more sense, especially when running `cli.getConfigStore().get(key);`

Reply:
Configstore is a specific implementation. I kept it generic to `config` in case we'd change the implementation in the future. That said, if we think that `Configstore` would be clearer, happy to adjust it.

Reply:
I suggest we keep it for now. It's an internal API that we can always revisit in the future without breaking anybody.